src/internal.c

```diff
@@ -6528,6 +6529,14 @@ static int DoCertificate(WOLFSSL* ssl, byte* input, word32* inOutIdx,
 
         InitDecodedCert(dCert, certs[0].buffer, certs[0].length, ssl->heap);
         ret = ParseCertRelative(dCert, CERT_TYPE, 0, ssl->ctx->cm);
+        if (ret != 0) {
+            WOLFSSL_MSG("ParseCertRelative failed");
+            FreeDecodedCert(dCert);
+        #ifdef WOLFSSL_SMALL_STACK
+            XFREE(dCert, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+        #endif
+            return ret;
```

Comment:
Please work with Jacob to fix this.  See next comment for why we probably can't just exit here.  But ret needs to be handled.